iceoryx_hoofs/test/moduletests/test_memory_bump_allocator.cpp

```diff
@@ -0,0 +1,146 @@
+    iox::BumpAllocator sut(memory, memorySize);
+    for (size_t i = 0; i < memorySize; i += 32)
+    {
+        auto allocationResult = sut.allocate(32, 1);
+        ASSERT_FALSE(allocationResult.has_error());
+        auto* bla = static_cast<size_t*>(allocationResult.value());
+        *bla = i;
+        EXPECT_THAT(*bla, Eq(i));
+    }
+}
+
+TEST_F(BumpAllocator_Test, allocateTooMuchSingleElement)
+{
+    ::testing::Test::RecordProperty("TEST_ID", "9deed5c0-19d8-4469-a5c3-f185d4d881f1");
+    iox::BumpAllocator sut(memory, memorySize);
+    auto allocationResult = sut.allocate(memorySize + 1, MEMORY_ALIGNMENT);
+    ASSERT_TRUE(allocationResult.has_error());
+    EXPECT_THAT(allocationResult.get_error(), Eq(iox::BumpAllocatorError::OUT_OF_MEMORY));
+}
+
+TEST_F(BumpAllocator_Test, allocateTooMuchMultipleElement)
+{
+    ::testing::Test::RecordProperty("TEST_ID", "435151e8-cc34-41ce-8115-5c179716a60a");
+    iox::BumpAllocator sut(memory, memorySize);
+    for (size_t i = 0; i < memorySize; i += 32)
+    {
+        ASSERT_FALSE(sut.allocate(32, 1).has_error());
+    }
+
+    auto allocationResult = sut.allocate(1, MEMORY_ALIGNMENT);
+    ASSERT_TRUE(allocationResult.has_error());
+    EXPECT_THAT(allocationResult.get_error(), Eq(iox::BumpAllocatorError::OUT_OF_MEMORY));
+}
+
+TEST_F(BumpAllocator_Test, allocateAndAlignment)
+{
+    ::testing::Test::RecordProperty("TEST_ID", "4252ddcc-05d4-499f-ad7c-30bffb420e08");
+    iox::BumpAllocator sut(memory, memorySize);
+    auto allocationResult = sut.allocate(5, MEMORY_ALIGNMENT);
+    ASSERT_FALSE(allocationResult.has_error());
+    auto* bla = static_cast<uint8_t*>(allocationResult.value());
+
+    allocationResult = sut.allocate(5, MEMORY_ALIGNMENT);
+    ASSERT_FALSE(allocationResult.has_error());
+    auto* bla2 = static_cast<uint8_t*>(allocationResult.value());
+    EXPECT_THAT(bla2 - bla, Eq(8U));
+}
+
+TEST_F(BumpAllocator_Test, allocateElementOfSizeZero)
+{
+    ::testing::Test::RecordProperty("TEST_ID", "17caa50c-94bf-4a1d-a1ec-dfda563caa0b");
+    iox::BumpAllocator sut(memory, memorySize);
+
+    // @todo iox-#1613 remove EXPECT_DEATH
+    // NOLINTBEGIN(hicpp-avoid-goto, cppcoreguidelines-avoid-goto, cert-err33-c, cppcoreguidelines-pro-type-vararg,
+    // hiccpp-vararg)
+    EXPECT_DEATH(IOX_DISCARD_RESULT(sut.allocate(0, MEMORY_ALIGNMENT)), ".*");
+    // NOLINTEND(hicpp-avoid-goto, cppcoreguidelines-avoid-goto, cert-err33-c, cppcoreguidelines-pro-type-vararg,
+    // hiccpp-vararg)
+}
+
+TEST_F(BumpAllocator_Test, allocateAfterDeallocateWorks)
+{
+    ::testing::Test::RecordProperty("TEST_ID", "323fc1af-481f-4732-b7d3-fa32da389cef");
+    iox::BumpAllocator sut(memory, memorySize);
+    ASSERT_FALSE(sut.allocate(memorySize, 1).has_error());
+
+    sut.deallocate();
+
+    auto allocationResult = sut.allocate(memorySize, 1);
+    ASSERT_FALSE(allocationResult.has_error());
+    int* bla = static_cast<int*>(allocationResult.value());
```

Comment:
We can fill the memory with something and check (but this was already done before, albeit without previous deallocation). It would be more important to check that we start with the "first" address before, i.e. the bump allocator is correctly reset.

Test name could be `deallocationResetsAllocator`.